v2/client/receiver.go

```diff
@@ -113,16 +113,16 @@ func (r *receiverFn) validateInParamSignature(fnType reflect.Type) error {
 	switch fnType.NumIn() {
 	case 2:
 		// has to be (context.Context, event.Event)
-		if !fnType.In(1).ConvertibleTo(eventType) {
-			return fmt.Errorf("%s; cannot convert parameter 2 from %s to event.Event", inParamUsage, fnType.In(1))
+		if !eventType.ConvertibleTo(fnType.In(1)) {
```

Comment:
Thinking about this more, the usecase is the function consumer would like to accept one of the interfaces that event.Event implements? Like even a subset? interesting. I had not considered this usage.

Reply:
Yeah this would for example use `event.EventReader` which exposes only part of an event. Not sure if this is useful, I changed this because older version was even less useful or less type safe.

Reply:
Again am not saying it is much useful, it is simply more type correct/safe, than having that `ConvertibleTo()` otherway around.